controllers/positions.js

```diff
@@ -1,80 +1,58 @@
+    .catch((error) => res.status(400).json(error));
 };
 
 const updatePosition = (req, res) => {
-  let updatedPosition;
-  const updatedPositions = Positions.map((position) => {
-    if (position.id === req.query.id) {
-      updatedPosition = {
-        id: req.query.id,
-        idClient: req.query.idClient,
-        jobDescription: req.query.jobDescription || position.jobDescription,
-        vacancy: req.query.vacancy || position.vacancy,
-        professionalProfiles: req.query.professionalProfiles || position.professionalProfiles,
-        status: req.query.status || position.status,
-      };
-      return updatedPosition;
-    }
-    return position;
-  });
+  Positions.findByIdAndUpdate(
+    req.params.id,
+    {
+      jobDescription: req.body.jobDescription,
+      vacancy: req.body.vacancy,
+      professionalProfiles: req.body.professionalProfiles,
+      isOpen: req.body.isOpen,
+    },
+    { new: true },
+    (error, newPosition) => {
+      if (!newPosition) {
+        return res.status(400).json({ msg: `Position with id: ${req.params.id} was not found` });
+      }
+      if (error) {
+        return res.status(400).json(error);
+      }
+      return res.status(200).json({ msg: 'Position updated', newPosition });
+    },
+  );
+};
 
-  if (!updatedPosition) {
-    return res.status(404).send('position not found');
-  }
-  if (JSON.stringify(updatedPositions) === JSON.stringify(Positions)) {
-    return res.status(200).send('position found but not changed');
-  }
+const createPosition = (req, res) => {
+  const positionCreated = new Positions({
+    client: req.body.client,
+    jobDescription: req.body.jobDescription,
+    vacancy: req.body.vacancy,
+    professionalProfiles: req.body.professionalProfiles,
+    isOpen: req.body.isOpen,
+  });
 
-  return fs.writeFile('./data/positions.json', JSON.stringify(updatedPositions), {}, (err) => {
-    if (err) {
-      return res.status(400).send(err);
+  positionCreated.save((error, position) => {
+    if (error) {
+      return res.status(400).json(error);
     }
-    return res.status(200).json(updatedPosition);
+    return res.status(201).json(position);
   });
 };
 
 const deletePosition = (req, res) => {
-  let removedPosition;
-  const filteredPositions = Positions.filter((app) => {
-    if (app.id === req.query.id) {
-      removedPosition = app;
-      return false;
+  Positions.findByIdAndDelete(req.params.id, (error) => {
+    if (error) {
+      return res.status(400).json(`Position with id ${req.params.id} does not extis`);
     }
-    return true;
-  });
-
-  if (!removedPosition.length) {
-    return res.status(404).send('Position not found');
-  }
-
-  return fs.writeFile('./data/positions.json', JSON.stringify(filteredPositions), {}, (err) => {
-    if (err) {
-      return res.status(400).send(err);
-    }
-    return res.status(204).json(removedPosition);
+    return res.status(204).json(`Position with id ${req.params.id} was deleted`);
```

Comment:
You should write 
```
return res.status(204).send();
```
because after status(204) no message is sent. You still need the .send(), if not the server doesn't send any response.